### PR TITLE
A few misc changes

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: csrc/scheduler/registry.cpp [hunk 1]
```diff
@@ -467,7 +467,7 @@ class SchedulerTopologyChecker {
       }
       // In the case of select, there's no consumer domain, and thus
       // there's no way to schedule the indexed producer domain
-      if (auto select = dynamic_cast<SelectOp*>(expr)) {
+      if (expr->isA<SelectOp>()) {
```
Comment (zasdfgbnm) [inline csrc/scheduler/registry.cpp:470]: This is a fix for gcc:
```
/usr/bin/g++-11 -DTORCH_CUDA_BUILD_MAIN_LIB -DUSE_C10D_GLOO -DUSE_C10D_MPI -DUSE_C10D_NCCL -DUSE_DISTRIBUTED -DUSE_NCCL_WITH_UCC -DUSE_RPC -DUSE_TENSORPIPE -Dnvfuser_codegen_EXPORTS -I/home/gaoxiang/Fuser/cmake/../third_party/benchmark/include -I/home/gaoxiang/Fuser/third_party/gloo -I/home/gaoxiang/Fuser/csrc -I/home/gaoxiang/Fuser/build/include -I/home/gaoxiang/Fuser/third_party/flatbuffers/include -isystem /home/gaoxiang/Fuser/cmake/../third_party/googletest/googlemock/include -isystem /home/gaoxiang/Fuser/cmake/../third_party/googletest/googletest/include -isystem /home/gaoxiang/Fuser/cmake/../third_party/flatbuffers/include -isystem /home/gaoxiang/cuda-11.8/include -isystem /home/gaoxiang/.local/lib/python3.10/site-packages/torch/include -isystem /home/gaoxiang/.local/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -D_GLIBCXX_USE_CXX11_ABI=1 -O3 -DNDEBUG -std=gnu++17 -fPIC -Wall -Wno-unused-function -DTORCH_CUDA_BUILD_MAIN_LIB -Werror -D_GLIBCXX_USE_CXX11_ABI=1 -MD -MT CMakeFiles/nvfuser_codegen.dir/csrc/scheduler/registry.cpp.o -MF CMakeFiles/nvfuser_codegen.dir/csrc/scheduler/registry.cpp.o.d -o CMakeFiles/nvfuser_codegen.dir/csrc/scheduler/registry.cpp.o -c /home/gaoxiang/Fuser/csrc/scheduler/registry.cpp
In file included from /home/gaoxiang/Fuser/csrc/ir_base_nodes.h:18,
                 from /home/gaoxiang/Fuser/csrc/disjoint_set.h:19,
                 from /home/gaoxiang/Fuser/csrc/scheduler/registry.cpp:9:
/home/gaoxiang/Fuser/csrc/utils.h: In instantiation of ‘bool nvfuser::PolymorphicBase::isA() const [with T = nvfuser::SelectOp*]’:
/home/gaoxiang/Fuser/csrc/scheduler/registry.cpp:470:31:   required from here
/home/gaoxiang/Fuser/csrc/utils.h:217:12: error: cannot ‘dynamic_cast’ ‘(const nvfuser::PolymorphicBase*)this’ (of type ‘const class nvfuser::PolymorphicBase*’) to type ‘class nvfuser::SelectOp* const*’ (target is not pointer or reference to class)
  217 |     return dynamic_cast<const T*>(this) != nullptr;
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[146/174] Building CXX object CMakeFiles/nvfuser.dir/csrc/python_frontend/python_bindings.cpp.o
```

File: test/utils.cpp [hunk 2]
```diff
@@ -479,4 +479,32 @@ bool isSchedulerInUse(
...
+  const auto& segment_groups = runtime->fusionSegments()->groups();
+
+  TORCH_CHECK(
+      segment_groups.size() == expected_heuristics.size(),
+      "Unexpected segments. Expected: ",
+      expected_heuristics.size(),
+      ". Actual: ",
+      segment_groups.size());
+
+  // Assumes up to two segments exist for simplicity
```
Comment (naoyam) [inline test/utils.cpp:494]: Please note about this constraint. It's enough for the cases I intended.

**Review by naoyam — APPROVED** [review]
LGTM

File: test/utils.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply test/utils.cpp:494]: Comment added

**Pull request merged**